packages/relay-compiler/core/RelayIR.js

```diff
@@ -158,7 +157,7 @@ export type ScalarField = {
   kind: 'ScalarField',
   metadata: ?{[key: string]: mixed},
   name: string,
-  type: GraphQLLeafType,
+  type: GraphQLOutputType,
```

Comment:
My concern is that this change makes it too easy to construct invalid IR. With the previous type, Flow would help catch errors in which you accidentally assigned an object type to a linked field.